File: Item.js
class Item {
  constructor(name) {
    this.name = name;
    this.id = this.generateUUID();
  }

  getName() {
    return this.name;
  }

  getId() {
    return this.id;
  }

  setName(name) {
    this.name = name;
  }

  generateUUID() {
    return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(
      /[xy]/g,
      function (c) {
        var r = (Math.random() * 16) | 0,
          v = c === "x" ? r : (r & 0x3) | 0x8;
        return v.toString(16);
      },
    );
  }
}

module.exports = Item;

File: src/main/java/org/example/model/Item.java
package org.example.model;

import java.util.UUID;

public class Item {
    private String id;
    private String name;

    public String getName() {
        return name;
    }

    public String getId() {
        return id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Item(String name) {
        this.name = name;
        this.id = UUID.randomUUID().toString();
    }
}
